lib/config.js

```diff
@@ -243,25 +243,41 @@ Config.prototype.getConfig = function(filePath) {
         config = ConfigOps.merge(config, this.useSpecificConfig);
     }
 
-    // Step 5: Merge in command line settings: environments, globals, parser,
-    // parserOptions, rules
-    debug("Merging command line settings");
-    config = ConfigOps.merge(config, {
-        env: this.env,
-        globals: this.globals,
-        parser: this.parser,
-        parserOptions: this.parserOptions,
-        rules: this.options.rules || {}
-    });
-
-    // Step 6: Merge in command line plugins
+    // Step 5: Merge in command line environments
+    debug("Merging command line environment settings");
+    config = ConfigOps.merge(config, { env: this.env });
+
+    // Step 6: Merge in command line rules
+    if (this.options.rules) {
+        debug("Merging command line rules");
+        config = ConfigOps.merge(config, { rules: this.options.rules });
+    }
+
+    // Step 7: Merge in command line globals
+    config = ConfigOps.merge(config, { globals: this.globals });
+
+    // Only override parser is is passed explicitly through command line or if its not
```

Comment:
Spelling: `// Only override parser if it is passed explicitly through the command line or if its not`